### PR TITLE
New board/artemis nano

### PR DESCRIPTION
**Pull request by AlienSarlak**
Resolved flashing issues, added documentation, and created new runner

- Fixed flashing issues for the board.
- Added necessary documentation.
- Created a new runner to support west flash on the board.

### CODE REVIEW

File: boards/sparkfun/artemis_nano/artemis_nano.dts [hunk 1]
```diff
@@ -0,0 +1,61 @@
+/dts-v1/;
+// ambiq_apollo3_blue.dtsi in dts/arm/ambiq
+#include <ambiq/ambiq_apollo3_blue.dtsi>
+//
```
Comment (bjarki-andreasen) [inline boards/sparkfun/artemis_nano/artemis_nano.dts:4]: `//` style comments are not allowed, use `/* */` :)

File: boards/sparkfun/artemis_nano/Kconfig.artemis_nano [hunk 2]
```diff
@@ -0,0 +1,10 @@
+# Sparkfun artemis nano board configuration
+# SPDX-License-Identifier: Apache-2.0
+
+
+config BOARD_ARTEMIS_NANO
+	select SOC_APOLLO3_BLUE
+
+config FLASH_BASE_ADDRESS
+    hex
+    default 0x10000
```
Comment (bjarki-andreasen) [inline boards/sparkfun/artemis_nano/Kconfig.artemis_nano:10]: There are missing newlines in some files :)

File: boards/sparkfun/artemis_nano/artemis_nano_defconfig [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+CONFIG_UART_CONSOLE=y
+CONFIG_SERIAL=y
+CONFIG_UART_INTERRUPT_DRIVEN=y
+
+# Enable GPIO support
+CONFIG_GPIO=y
+
+# Enable LED support
+CONFIG_LED=y
+CONFIG_LED_GPIO=y
```
Comment (bjarki-andreasen) [inline boards/sparkfun/artemis_nano/artemis_nano_defconfig:15]: Including drivers should be done from the application level, the basic blinky sample actually just uses the GPIO device driver directly, and `CONFIG_GPIO` is selected by the sample :)

The only drivers commonly included here are the ones required for the console, so even GPIO may be left out here :)

**Review by bjarki-andreasen — CHANGES_REQUESTED** [review]

File: boards/sparkfun/artemis_nano/Kconfig.artemis_nano (hunk 2 above)
Comment (AlienSarlak) [reply boards/sparkfun/artemis_nano/Kconfig.artemis_nano:10]: Could you please explain it more for me. What do you mean by "missing newlines"?

File: boards/sparkfun/artemis_nano/artemis_nano_defconfig (hunk 3 above)
Comment (AlienSarlak) [reply boards/sparkfun/artemis_nano/artemis_nano_defconfig:15]: Understood. Thanks

File: boards/sparkfun/artemis_nano/Kconfig.artemis_nano (hunk 2 above)
Comment (bjarki-andreasen) [reply boards/sparkfun/artemis_nano/Kconfig.artemis_nano]: Files need to end with a newline, in most IDEs, the end of a file should look like
```
132 |         foo();
133 | }
134
```
You can see the files missing a newline if you review this PR on github https://github.com/zephyrproject-rtos/zephyr/pull/75210/files

**Comment by bjarki-andreasen** [comment]
You will need to squash commits which undo previous commits and remove merges. The PR should be based on upstream/main and every commit after must be based in the previous commit.

**Pull request closed without merging**